components/DropdownBtn.tsx

```diff
@@ -1,72 +1,101 @@
-import { Menu } from '@headlessui/react'
+import React from 'react'
+import { Menu, Transition } from '@headlessui/react'
 import { ChevronDownIcon } from '@heroicons/react/solid'
+import { InformationCircleIcon } from '@heroicons/react/outline'
 import Loading from '@components/Loading'
+import Tooltip from './Tooltip'
 
-export interface DropdownBtnOptions {
-  isDefault: boolean | undefined
+interface DropdownOption {
   label: string
-  callback: () => Promise<void>
+  disabled?: boolean
+  disabledTooltip?: string
+  onClick: () => Promise<void>
 }
 
-const DropdownBtn = ({
-  options,
-  isLoading,
-}: {
-  options: DropdownBtnOptions[]
+interface DropdownBtnProps {
+  className: string
+  label?: string
+  options: DropdownOption[]
   isLoading?: boolean
+  disabled?: boolean
+}
+
+const DropdownBtn: React.FC<DropdownBtnProps> = ({
+  className,
+  label = 'Options',
+  options,
+  isLoading = false,
+  disabled = false,
 }) => {
-  const defaultFunction = options.find((x) => x.isDefault)
-  if (!defaultFunction) {
-    throw 'DropdownBtn must have at least one default option'
-  }
-  const filtredOptions = options.filter((x) => !x.isDefault)
+  const basicClasses = `inline-flex rounded-full w-full justify-center bg-primary-light text-bkg-2 px-4 py-3 text-sm font-medium`
+  const hoverClasses = `hover:bg-opacity-30 focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75 hover:bg-fgd-1`
+  const disabledClasses = disabled
+    ? `!bg-fgd-4 !cursor-not-allowed !text-fgd-3`
+    : ''
+  const btnClasses = `${basicClasses} ${hoverClasses} ${disabledClasses}`
```

Comment:
nit: hmm. my ide has autocomplete and other nice features for tailwind classes. Instead of breaking these up in this manner, I would suggest simply having the classnames inline, but break them up with line breaks and comments.

Reply:
on the other hand i don't care so you can ignore this. this component is surely not long for this world anyway.